### ISSUE TITLE
Transforms and samplers order in queue / transformrs after sampling 

### ISSUE THREAD

**Issue by PeterMcGor** [fepegar/torchio#445]
**🚀 Feature**

Hey again!

In the current version, the order of transforms and sampling in the queue is fixed as is clearly explained [here](https://torchio.readthedocs.io/data/patch_based.html). 
Sometimes, transformations need to be done (or obviously present different results when are applied on...) on the patches instead of the original image, although, there some workarounds, it could be nice to have some built-in option within the `Queue` object to specify some transformation after the patch creation. 

**Motivation**
Allow defining patch-based transformations

**Alternatives**

I have been looking through old issues but I could not find something related, probably I am wrong so please let me know in that case. 

Best!

**Comment by fepegar** [comment]
Hi, @PeterMcGor.

Which transforms would you like to use? I suppose a `patches_transform` could be passed to the `Queue`. However, it wouldn't be applied using multiprocessing (I think).

**Comment by PeterMcGor** [comment]
Hi!

For example, intensity transformations based on the patch stats, or spatial ones. I guess a best example could be an 2D architecture which employ 2D patches. Depending on the setup, some works standardise/normalise the values taking the whole datasets, the volume or the slice.  
Since right now torchio pop patches from a list obviously the multiprocessing is not possible (by just adding a set of transformation after patch extraction). There are some ways to workaround that are closely related with the hot topic about `Queue` and GPU utilisation #393. Looks like we are all employing dirty workarounds but none of us have the time to properly solve it... I will try to keep tabs on this. 

Cheers!

**Comment by fepegar** [comment]
If it's for 2D patches and speed is not super critical, you might get away with applying the transform with a single process:

```python
In [1]: import torchio as tio

In [2]: transform = tio.Compose([tio.RescaleIntensity(), tio.RandomFlip((0, 1))])

In [3]: import torch

In [4]: image = tio.ScalarImage(tensor=torch.rand(1, 256, 256, 1))

In [5]: %timeit transform(image)
1.32 ms ± 54.7 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [6]: transform = tio.Compose([tio.RescaleIntensity(), tio.RandomFlip((0, 1)), tio.RandomElasticDeformation()])

In [7]: %timeit transform(image)
54.5 ms ± 3.03 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [8]: transform = tio.Compose([tio.RescaleIntensity(), tio.RandomFlip((0, 1)), tio.RandomAffine()])

In [9]: %timeit transform(image)
4.54 ms ± 426 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

```

**Comment by fepegar** [comment]
Closing this unless there is a specific request. Feel free to reopen.
